### DISCUSSION TITLE
Requests left hanging

### DISCUSSION THREAD

**Discussion by uSkizzik** [chimurai/http-proxy-middleware#794] [Q&A] [answered]
Aight so, here's a quick summary:  
Essentially all of my requests to the proxy server get no responses. I've found out that requests with no bodies actually do get responses, the others don't. I've tried sending the request directly to the server rather than the proxy and everything works just fine.  

Another thing - if I try printing to console in the request handler for an endpoint, nothing gets printed (again, only if the request was sent to the proxy rather than directly to the server)

This is my setup for the proxy:
```js
const app = express()
const router = express.Router({})

app.use("/api", router)
router.use(express.json())

router.use(
	"/auth",
	createProxyMiddleware({
		target: "http://localhost:8082",
		changeOrigin: true
	})
)

const server = http.createServer(app)
server.listen(port, () => {
	console.log(`Server running on port ${port}`)
})
```

**Comment by uSkizzik** [comment, marked as answer]
The issue was this line:
```js
router.use(express.json())
```
Removing it fixed the issue and everything works like a charm.

**Reply by Miggets7** [reply]
Can you elaborate how this fixes hanging of the requests?
